src/test/kotlin/org/rust/ide/annotator/RsExpressionAnnotatorTest.kt

```diff
@@ -98,4 +98,18 @@ class RsExpressionAnnotatorTest : RsAnnotatorTestBase() {
             let _ = U <error descr="Union expressions should have exactly one field">{ a: 92, b: 92.0}</error>;
         }
     """)
+
+    fun testReassignImmutableFromBinding() = checkErrors("""
+        fn main() {
+            let x = 5;
+            <error descr="Reassigning an immutable variable [E0384]">x = 3</error>;
+        }
+    """)
+
+    fun testReassignMutableFromBinding() = checkErrors("""
+        fn main() {
+            let mut x = 5;
+            x = 3;
+        }
+    """)
```

Comment:
And I would also add a couple of tests to guarantee regression checks. For instance, a test for `let` without assignment, a test for assignment through asterisk (we should ignore them).